extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/PackageUtil.java

```diff
@@ -0,0 +1,216 @@
+import io.dekorate.deps.commons.compress.archivers.tar.TarArchiveOutputStream;
+
+public class PackageUtil {
+
+    private static final String DEFAULT_DOCKERFILE = "Dockerfile";
+    private static final String DOCKER_IGNORE = ".dockerignore";
+
+    protected static final String DEFAULT_TEMP_DIR = System.getProperty("java.io.tmpdir", "/tmp");
+    protected static final String DOCKER_PREFIX = "docker-";
+    protected static final String BZIP2_SUFFIX = ".tar.bzip2";
+
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+    public static File packageFile(String path) {
+        return packageFile(path, null);
+    }
+
+    public static File packageFile(String path, String base) {
+        try {
+            final Path root = Paths.get(path).getParent();
+            File tempFile = Files.createTempFile(Paths.get(DEFAULT_TEMP_DIR), DOCKER_PREFIX, BZIP2_SUFFIX).toFile();
+            try (final TarArchiveOutputStream tout = buildTarStream(tempFile)) {
+                Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+                        return FileVisitResult.CONTINUE;
+                    }
+
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        String absolutePath = file.toAbsolutePath().toString();
+                        if (!path.equals(absolutePath)) {
+                            return FileVisitResult.CONTINUE;
+                        }
+                        final Path relativePath = root.relativize(file);
+                        final boolean hasBasePath = !StringUtil.isBlank(base);
+                        final TarArchiveEntry entry = hasBasePath
+                                ? new TarArchiveEntry(base + File.separator + file.toFile())
+                                : new TarArchiveEntry(file.toFile());
+
+                        entry.setName(hasBasePath ? base + File.separator + relativePath.toString() : relativePath.toString());
+                        if (file.toFile().canExecute()) {
+                            entry.setMode(entry.getMode() | 0755);
+                        }
+                        entry.setSize(attrs.size());
+                        putTarEntry(tout, entry, file);
+                        return FileVisitResult.CONTINUE;
+                    }
+                });
+                tout.flush();
+            }
+            return tempFile;
+
+        } catch (IOException e) {
+            throw DekorateException.launderThrowable(e);
+        }
+    }
+
+    public static File packageFile(Path root, Path... additional) {
+        return packageFile(root, null, additional);
+    }
+
+    public static File packageFile(Path root, String base, Path... additional) {
```

Comment:
This method seems to have plenty of similarities with `packageFile`. Can the common code be extracted?